### ISSUE TITLE
how to require i18next in isomorphic code?

### ISSUE THREAD

**Issue by i524** [i18next/i18next#1334]
Hello!
In i18next@17.0.18, `dist/esm/i18next.js` and `dist/cjs/i18next.js` exports things in different ways;

`dist/esm/i18next.js` uses default export

```javascript
// the last line of dist/esm/i18next.js
export default i18next;
```

and `dist/cjs/i18next.js` doesn't

```javascript
// the last line of dist/cjs/i18next.js
module.exports = i18next;
```

So I got to do `const i18n = require('i18next')` for node.js environment and `const i18n = require('i18next').default` for client side code (using webpack).

Is this a intended design?
If so, how should I require() i18next in isomorphic codes?

**Comment by jamuhl** [comment]
`const i18n = require('i18next/dist/cjs/i18next')`

**Comment by jamuhl** [comment]
but where do you run into an isomorphic issue...? never had such problems...

**Comment by i524** [comment]
> const i18n = require('i18next/dist/cjs/i18next')

That will work but I think esm version should be used for webpack.

> but where do you run into an isomorphic issue...? never had such problems...

I'll try to make a repro

**Comment by i524** [comment]
@jamuhl I created a repro!
https://github.com/i524/issue-repro-i18next-1334

When you access localhost:3000 you will see an error page saying `i18n.init is not a function`, but on the server-side i18next is working correctly (as you can see in console)

**Comment by jamuhl** [comment]
To not waste to much time...why not head over to: https://github.com/isaachinman/next-i18next

proven to work...and is full fledged solution for next.js

**Comment by jamuhl** [comment]
And doing:

```js
const i18next = require("i18next");

const i18n = i18next.default || i18next;
```

should do the trick for this scenario

**Comment by i524** [comment]
@jamuhl sorry for the delay

>> why not head over to: https://github.com/isaachinman/next-i18next

The project I work on is not a next.js project.
It's a bit complicated code so I made a simpler one with next.js(I should have mentioned that, sorry).

I read some codes in react-i18next and It turned out that when you use import syntax and babel, babel actually does that trick(`i18next.default || i18next`). I didn't know this...

So I should use import syntax or your trick. I got it.
Thank you for helping! closing this.
